base/linalg/matmul.jl

```diff
@@ -44,7 +44,7 @@ end
 vecdot{T<:BlasReal}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dot(x, y)
 vecdot{T<:BlasComplex}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dotc(x, y)
 
-function dot{T<:BlasReal, TI<:Integer}(x::Vector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::Vector{T}, ry::Union{UnitRange{TI},Range{TI}})
+function dot{T<:BlasReal, TI<:Integer}(x::StridedVector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::StridedVector{T}, ry::Union{UnitRange{TI},Range{TI}})
```

Comment:
> Maybe we shouldn't allow both that the vectors are strided and a range argument. It seems kind of double to me. Ideally these methods should actually be deleted and we should just use subarrays but there is probably still a price to pay over using this approach.

@andreasnoack Do you mean sub-ing according to the input ranges, and calling the `StridedVector` version of `dot(c)`? I tried that, but there's a substantial relative time increase. https://gist.github.com/pkofod/901639371f4c05925e171dd279147f16